tv_apps/smart-system/js/splash_screen.js

```diff
@@ -56,5 +63,45 @@
     this.element.classList.add('hidden');
   };
 
+  SplashScreen.prototype.getIconUrl = function ss_getIconUrl(app) {
+    if (app.manifestURL) {
+      return this.getIconFromManifestURL(app.manifestURL);
+    } else if (app.features.iconUrl) {
+      return app.features.iconUrl;
+    }
+  };
+
+  SplashScreen.prototype.getIconFromManifestURL = function(manifestURL) {
+    var app = applications.getByManifestURL(manifestURL);
+    var icons = app.manifest.icons;
+
+    var maxIconSize = Object.keys(icons).reduce(function(prev, current) {
+        return parseInt(prev, 10) < parseInt(current, 10) ? current : prev;
+    });
+
+    var url = icons[maxIconSize];
+
+    if (!url) {
+      return;
+    }
+
+    if (url.indexOf('data:') === 0 ||
+        url.indexOf('app://') === 0 ||
+        url.indexOf('http://') === 0 ||
+        url.indexOf('https://') === 0) {
```

Comment:
I would use `startsWith()` [1] or `match(regexp)` [2] to simplify these conditions, but it's up to you.

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith
[2] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match